src/main/java/org/isf/accounting/rest/BillController.java

```diff
@@ -131,9 +131,9 @@ public BillController(BillBrowserManager billManager, PriceListManager priceList
         	throw new OHAPIException(new OHExceptionMessage(null, "Price list not found!", OHSeverityLevel.ERROR));
         }
         
-        ArrayList<BillItems> billItems = new ArrayList<BillItems>(billItemsMapper.map2ModelList(newBillDto.getBillItemsDTO()));
+        ArrayList<BillItems> billItems = new ArrayList<BillItems>(billItemsMapper.map2ModelList(newBillDto.getBillItems()));
```

Comment:
Not clear to me: if we want to rename the `billItems` property into `billItemsDTO` will Lombok update also the related getters and setters automatically?

Reply:
Yes that is true but most of the methods/properties did not use the "DTO" part so I was making them consistent.

Reply:
yes the intention was clear, I was just asking how Lombok works and helps (also) in refactoring because I don't know it 

(TBH I tried to refactor `billItems`->`billItemsDTO` in IDE Eclipse and getters in [R134](https://github.com/informatici/openhospital-api/pull/82/files#diff-0c44ffb6e41726bcf884d8c344783d40884e5470a453adf29e74bb8bd63dbabaR185) is not updated accordingly)

Reply:
The methods that are called are not changed automatically.  When the Lombok annotations are added behind the scenes the `getXXX`, `setXXX`, and maybe `isXXX` methods are generated.   Any place that calls the methods are not automatically updated.   That is still a manual process.

Reply:
I see... I think Lombok should cover this because otherwise the good of having removed boilerplate code (that before you was writing once and for all) is diminished by the absence of an important feature in IDEs that is the one of refactoring properties and methods automatically (and not manually). And it seems I'm foreshadowing what other people already experienced, like also the "[find references](https://stackoverflow.com/q/31675937/5309541)" feature (even more useful indeed).

Maybe we can suggest/document also the workarounds proposed by those same people?

Reply:
You are enumerating the reasons that I don't like Lombok.  All it does really is not have getters/setters in your code.   Given that most modern IDE's will automatically generate the code (and often with Javadoc) it really isn't saving any typing.   It is just too much (partial) magic for my taste.   If you haven't figured it out by now I tend to have opinions on many topics :)

Reply:
I'll be more than happy to undo the changes and create a new PR.

Reply:
Actually I was not deciding [alone that](https://github.com/informatici/openhospital-api/pull/16), but I totally agree with you now that I know what Lombok is and does.
Let me ask @pviotti if agrees to undo changes and also remove the plugin?

Reply:
@pviotti Are you OK with not using Lombok?

Reply:
fine by me. I'd really love to remove some boilerplate code but if it comes at the price of hindering features of popular IDEs, then maybe not.

Reply:
OK; then the decision appears to be to remove Lombok; I'll create a new PR tomorrow that restores the logging and the `boilerplate` code.  Simple and straightforward is always the best approach IMHO.

Reply:
Remove Lombok!!!  interesting and kind of radical solution :) 

@dbmalkovsky  @pviotti  @mwithi Would it be a temporal solution?

I am windering how the progresses about Lombok adoption is in the other modules CORE and GUI ?

Reply:
Just pushed the changes to remove Lombok in this project.

@mwithi I also mad the change to the method name as you suggested; now using `patientTrue`.

Reply:
@emecas This appears to be the only project that started to use Lombok.   The cost benefit ratio for using Lombok is just too high (IMHO) if you want to do easy refactoring and have complete control over project.

Reply:
> @mwithi I also made the change to the method name as you suggested; now using `patientTrue`.

Thanks a lot!

> if you want to do easy refactoring and have complete control over project.

yeah, and it seems a forgotten issues on their [roadmap](https://groups.google.com/g/project-lombok/search?q=refactoring)